<Type Name="UnhandledExceptionAction" FullName="System.Activities.UnhandledExceptionAction">
  <Metadata><Meta Name="ms.openlocfilehash" Value="5ef7a0a32b99e9ac13071b3e26f38aa2e7ad640e" /><Meta Name="ms.sourcegitcommit" Value="f1d16425528e237257ca3b58eb49217a514849ea" /><Meta Name="ms.translationtype" Value="MT" /><Meta Name="ms.contentlocale" Value="pl-PL" /><Meta Name="ms.lasthandoff" Value="04/24/2019" /><Meta Name="ms.locfileid" Value="63917111" /></Metadata><TypeSignature Language="C#" Value="public enum UnhandledExceptionAction" />
  <TypeSignature Language="ILAsm" Value=".class public auto ansi sealed UnhandledExceptionAction extends System.Enum" />
  <TypeSignature Language="DocId" Value="T:System.Activities.UnhandledExceptionAction" />
  <TypeSignature Language="VB.NET" Value="Public Enum UnhandledExceptionAction" />
  <TypeSignature Language="C++ CLI" Value="public enum class UnhandledExceptionAction" />
  <TypeSignature Language="F#" Value="type UnhandledExceptionAction = " />
  <AssemblyInfo>
    <AssemblyName>System.Activities</AssemblyName>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <Base>
    <BaseTypeName>System.Enum</BaseTypeName>
  </Base>
  <Docs>
    <summary><span data-ttu-id="be915-101">Określa akcję, która występuje, gdy wyjątek specjalne głównego przepływu pracy.</span><span class="sxs-lookup"><span data-stu-id="be915-101">Specifies the action that occurs when an exception escapes the root of a workflow.</span></span></summary>
    <remarks>
      <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="be915-102"><xref:System.Activities.WorkflowApplication.OnUnhandledException%2A> Funkcja jest wywoływana, jeśli wyjątek specjalne głównego przepływu pracy.</span><span class="sxs-lookup"><span data-stu-id="be915-102">The <xref:System.Activities.WorkflowApplication.OnUnhandledException%2A> function is invoked if an exception escapes the root of the workflow.</span></span> <span data-ttu-id="be915-103"><xref:System.Activities.WorkflowApplicationUnhandledExceptionEventArgs> Zapewnia dostęp do tego wyjątku, a także wskaźnik do <xref:System.Activities.Activity> , wygenerowany wyjątek.</span><span class="sxs-lookup"><span data-stu-id="be915-103">The <xref:System.Activities.WorkflowApplicationUnhandledExceptionEventArgs> provides access to the exception as well as a pointer to the <xref:System.Activities.Activity> that generated the exception.</span></span> <span data-ttu-id="be915-104">Zakończenie jest domyślną akcję, jeśli nie <xref:System.Activities.WorkflowApplication.OnUnhandledException%2A> program obsługi jest określony.</span><span class="sxs-lookup"><span data-stu-id="be915-104">Terminate is the default action if no <xref:System.Activities.WorkflowApplication.OnUnhandledException%2A> handler is specified.</span></span>  
  
   
  
## Examples  
 <span data-ttu-id="be915-105">Poniższy przykład przedstawia wywoływanie przepływu pracy, która zgłosiła wyjątek.</span><span class="sxs-lookup"><span data-stu-id="be915-105">The following example invokes a workflow that throws an exception.</span></span> <span data-ttu-id="be915-106">Wyjątek jest nieobsługiwany przez przepływ pracy i <xref:System.Activities.WorkflowApplication.OnUnhandledException%2A> zostanie wywołana procedura obsługi.</span><span class="sxs-lookup"><span data-stu-id="be915-106">The exception is unhandled by the workflow and the <xref:System.Activities.WorkflowApplication.OnUnhandledException%2A> handler is invoked.</span></span> <span data-ttu-id="be915-107"><xref:System.Activities.WorkflowApplicationUnhandledExceptionEventArgs> Są kontrolowane w celu dostarczenia informacji o wyjątku, a przepływ pracy zostanie zakończony.</span><span class="sxs-lookup"><span data-stu-id="be915-107">The <xref:System.Activities.WorkflowApplicationUnhandledExceptionEventArgs> are inspected to provide information about the exception, and the workflow is terminated.</span></span>  
  
 [!code-csharp[CFX_WorkflowApplicationExample#6](~/samples/snippets/csharp/VS_Snippets_CFX/cfx_workflowapplicationexample/cs/program.cs#6)]  
  
 ]]></format>
    </remarks>
  </Docs>
  <Members>
    <Member MemberName="Abort">
      <MemberSignature Language="C#" Value="Abort" />
      <MemberSignature Language="ILAsm" Value=".field public static literal valuetype System.Activities.UnhandledExceptionAction Abort = int32(0)" />
      <MemberSignature Language="DocId" Value="F:System.Activities.UnhandledExceptionAction.Abort" />
      <MemberSignature Language="VB.NET" Value="Abort" />
      <MemberSignature Language="C++ CLI" Value="Abort" />
      <MemberSignature Language="F#" Value="Abort = 0" Usage="System.Activities.UnhandledExceptionAction.Abort" />
      <MemberType>Field</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Activities</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <ReturnValue>
        <ReturnType>System.Activities.UnhandledExceptionAction</ReturnType>
      </ReturnValue>
      <MemberValue>0</MemberValue>
      <Docs>
        <summary><span data-ttu-id="be915-108">Określa, że <see cref="T:System.Activities.WorkflowApplication" /> powinien przerwać przepływu pracy.</span><span class="sxs-lookup"><span data-stu-id="be915-108">Specifies that the <see cref="T:System.Activities.WorkflowApplication" /> should abort the workflow.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="be915-109">Skutkuje to <xref:System.Activities.WorkflowApplication.Aborted%2A> wywoływana po zakończeniu procesu przerwania.</span><span class="sxs-lookup"><span data-stu-id="be915-109">This results in <xref:System.Activities.WorkflowApplication.Aborted%2A> being called when the abort process is complete.</span></span> <span data-ttu-id="be915-110">Nieobsługiwany wyjątek jest używana jako powód przerwania.</span><span class="sxs-lookup"><span data-stu-id="be915-110">The unhandled exception is used as the abort reason.</span></span>  
  
   
  
## Examples  
 <span data-ttu-id="be915-111">Poniższy przykład przedstawia wywoływanie przepływu pracy, która zgłosiła wyjątek.</span><span class="sxs-lookup"><span data-stu-id="be915-111">The following example invokes a workflow that throws an exception.</span></span> <span data-ttu-id="be915-112">Wyjątek jest nieobsługiwany przez przepływ pracy i <xref:System.Activities.WorkflowApplication.OnUnhandledException%2A> zostanie wywołana procedura obsługi.</span><span class="sxs-lookup"><span data-stu-id="be915-112">The exception is unhandled by the workflow and the <xref:System.Activities.WorkflowApplication.OnUnhandledException%2A> handler is invoked.</span></span> <span data-ttu-id="be915-113"><xref:System.Activities.WorkflowApplicationUnhandledExceptionEventArgs> Są kontrolowane w celu dostarczenia informacji o wyjątku, i przepływu pracy zostało przerwane.</span><span class="sxs-lookup"><span data-stu-id="be915-113">The <xref:System.Activities.WorkflowApplicationUnhandledExceptionEventArgs> are inspected to provide information about the exception, and the workflow is aborted.</span></span>  
  
 [!code-csharp[CFX_WorkflowApplicationExample#7](~/samples/snippets/csharp/VS_Snippets_CFX/cfx_workflowapplicationexample/cs/program.cs#7)]  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName="Cancel">
      <MemberSignature Language="C#" Value="Cancel" />
      <MemberSignature Language="ILAsm" Value=".field public static literal valuetype System.Activities.UnhandledExceptionAction Cancel = int32(1)" />
      <MemberSignature Language="DocId" Value="F:System.Activities.UnhandledExceptionAction.Cancel" />
      <MemberSignature Language="VB.NET" Value="Cancel" />
      <MemberSignature Language="C++ CLI" Value="Cancel" />
      <MemberSignature Language="F#" Value="Cancel = 1" Usage="System.Activities.UnhandledExceptionAction.Cancel" />
      <MemberType>Field</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Activities</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <ReturnValue>
        <ReturnType>System.Activities.UnhandledExceptionAction</ReturnType>
      </ReturnValue>
      <MemberValue>1</MemberValue>
      <Docs>
        <summary><span data-ttu-id="be915-114">Określa, że <see cref="T:System.Activities.WorkflowApplication" /> należy zaplanować anulowania działania głównego i wznowić wykonywanie.</span><span class="sxs-lookup"><span data-stu-id="be915-114">Specifies that the <see cref="T:System.Activities.WorkflowApplication" /> should schedule the cancellation of the root activity and resume execution.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="be915-115">Skutkuje to <xref:System.Activities.WorkflowApplication.Completed%2A> wywoływana po zakończeniu procesu anulowania.</span><span class="sxs-lookup"><span data-stu-id="be915-115">This results in <xref:System.Activities.WorkflowApplication.Completed%2A> being called when the cancellation process is complete.</span></span>  
  
   
  
## Examples  
 <span data-ttu-id="be915-116">Poniższy przykład przedstawia wywoływanie przepływu pracy, która zgłosiła wyjątek.</span><span class="sxs-lookup"><span data-stu-id="be915-116">The following example invokes a workflow that throws an exception.</span></span> <span data-ttu-id="be915-117">Wyjątek jest nieobsługiwany przez przepływ pracy i <xref:System.Activities.WorkflowApplication.OnUnhandledException%2A> zostanie wywołana procedura obsługi.</span><span class="sxs-lookup"><span data-stu-id="be915-117">The exception is unhandled by the workflow and the <xref:System.Activities.WorkflowApplication.OnUnhandledException%2A> handler is invoked.</span></span> <span data-ttu-id="be915-118"><xref:System.Activities.WorkflowApplicationUnhandledExceptionEventArgs> Są kontrolowane w celu dostarczenia informacji o wyjątku, i przepływu pracy zostało anulowane.</span><span class="sxs-lookup"><span data-stu-id="be915-118">The <xref:System.Activities.WorkflowApplicationUnhandledExceptionEventArgs> are inspected to provide information about the exception, and the workflow is cancelled.</span></span>  
  
 [!code-csharp[CFX_WorkflowApplicationExample#8](~/samples/snippets/csharp/VS_Snippets_CFX/cfx_workflowapplicationexample/cs/program.cs#8)]  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName="Terminate">
      <MemberSignature Language="C#" Value="Terminate" />
      <MemberSignature Language="ILAsm" Value=".field public static literal valuetype System.Activities.UnhandledExceptionAction Terminate = int32(2)" />
      <MemberSignature Language="DocId" Value="F:System.Activities.UnhandledExceptionAction.Terminate" />
      <MemberSignature Language="VB.NET" Value="Terminate" />
      <MemberSignature Language="C++ CLI" Value="Terminate" />
      <MemberSignature Language="F#" Value="Terminate = 2" Usage="System.Activities.UnhandledExceptionAction.Terminate" />
      <MemberType>Field</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Activities</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <ReturnValue>
        <ReturnType>System.Activities.UnhandledExceptionAction</ReturnType>
      </ReturnValue>
      <MemberValue>2</MemberValue>
      <Docs>
        <summary><span data-ttu-id="be915-119">Określa, że <see cref="T:System.Activities.WorkflowApplication" /> należy zaplanować zakończenie działania głównego i wznowić wykonywanie.</span><span class="sxs-lookup"><span data-stu-id="be915-119">Specifies that the <see cref="T:System.Activities.WorkflowApplication" /> should schedule termination of the root activity and resume execution.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="be915-120">Skutkuje to <xref:System.Activities.WorkflowApplication.Completed%2A> wywoływana po zakończeniu procesu zakończenia.</span><span class="sxs-lookup"><span data-stu-id="be915-120">This results in <xref:System.Activities.WorkflowApplication.Completed%2A> being called when the termination process is complete.</span></span> <span data-ttu-id="be915-121">Nieobsługiwany wyjątek jest używana jako Przyczyna zakończenia.</span><span class="sxs-lookup"><span data-stu-id="be915-121">The unhandled exception is used as the termination reason.</span></span> <span data-ttu-id="be915-122">Zakończenie jest domyślną akcję, jeśli nie <xref:System.Activities.WorkflowApplication.OnUnhandledException%2A> program obsługi jest określony.</span><span class="sxs-lookup"><span data-stu-id="be915-122">Terminate is the default action if no <xref:System.Activities.WorkflowApplication.OnUnhandledException%2A> handler is specified.</span></span>  
  
   
  
## Examples  
 <span data-ttu-id="be915-123">Poniższy przykład przedstawia wywoływanie przepływu pracy, która zgłosiła wyjątek.</span><span class="sxs-lookup"><span data-stu-id="be915-123">The following example invokes a workflow that throws an exception.</span></span> <span data-ttu-id="be915-124">Wyjątek jest nieobsługiwany przez przepływ pracy i <xref:System.Activities.WorkflowApplication.OnUnhandledException%2A> zostanie wywołana procedura obsługi.</span><span class="sxs-lookup"><span data-stu-id="be915-124">The exception is unhandled by the workflow and the <xref:System.Activities.WorkflowApplication.OnUnhandledException%2A> handler is invoked.</span></span> <span data-ttu-id="be915-125"><xref:System.Activities.WorkflowApplicationUnhandledExceptionEventArgs> Są kontrolowane w celu dostarczenia informacji o wyjątku, a przepływ pracy zostanie zakończony.</span><span class="sxs-lookup"><span data-stu-id="be915-125">The <xref:System.Activities.WorkflowApplicationUnhandledExceptionEventArgs> are inspected to provide information about the exception, and the workflow is terminated.</span></span>  
  
 [!code-csharp[CFX_WorkflowApplicationExample#6](~/samples/snippets/csharp/VS_Snippets_CFX/cfx_workflowapplicationexample/cs/program.cs#6)]  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
  </Members>
</Type>